Script/update_grovepi.sh

```diff
@@ -5,34 +5,147 @@ PIHOME=/home/pi
+# whether to install the dependencies or not (avrdude, apt-get, wiringpi, and so on)
+installdependencies=true
+updaterepo=true
+install_pkg_scriptools=true
 
-GROVEPI_DIR=$DEXTER_PATH/GrovePi
+# the following 3 options are mutually exclusive
+systemwide=true
+userlocal=false
+envlocal=false
+usepython3exec=true
+
+# the following option tells which branch has to be used
+selectedbranch="master"
 
-# Check for a GrovePi directory under "Dexter" folder.  If it doesn't exist, create it.
+declare -a optionslist=("--system-wide")
+# iterate through bash arguments
+for i; do
+  case "$i" in
+    --no-dependencies)
+      installdependencies=false
+      ;;
+    --no-update-aptget)
+      updaterepo=false
+      ;;
+    --bypass-pkg-scriptools)
+      install_pkg_scriptools=false
+      ;;
+    --user-local)
+      userlocal=true
+      systemwide=false
+      declare -a optionslist=("--user-local")
+      ;;
+    --env-local)
+      envlocal=true
+      systemwide=false
+      declare -a optionslist=("--env-local")
+      ;;
+    --system-wide)
+      ;;
+    develop|feature/*|hotfix/*|fix/*|DexterOS*|v*)
+      selectedbranch="$i"
+      ;;
+  esac
+done
 
-if [ -d "$GROVEPI_DIR" ]; then
-    echo "GrovePi Directory Exists"
-    cd $GROVEPI_DIR             # Go to directory
-    sudo git fetch origin       # Hard reset the git files
-    sudo git reset --hard  
-    sudo git merge origin/master
+# show some feedback for the GrovePi
+echo "  _____            _                                ";
+echo " |  __ \          | |                               ";
+echo " | |  | | _____  _| |_ ___ _ __                     ";
+echo " | |  | |/ _ \ \/ / __/ _ \ '__|                    ";
+echo " | |__| |  __/>  <| ||  __/ |                       ";
+echo " |_____/ \___/_/\_\\\__\___|_|          _            ";
+echo " |_   _|         | |         | |      (_)           ";
+echo "   | |  _ __   __| |_   _ ___| |_ _ __ _  ___  ___  ";
+echo "   | | | '_ \ / _\ | | | / __| __| '__| |/ _ \/ __| ";
+echo "  _| |_| | | | (_| | |_| \__ \ |_| |  | |  __/\__ \ ";
+echo " |_____|_| |_|\__,_|\__,_|___/\__|_|  |_|\___||___/ ";
+echo "                                                    ";
+echo "                                                    ";
+echo "  _____                    _____ _ "
+echo " / ____|                  |  __ (_)  "
+echo "| |  __ _ __ _____   _____| |__) |   "
+echo "| | |_ | '__/ _ \ \ / / _ \  ___/ |  "
+echo "| |__| | | | (_) \ V /  __/ |   | |  "
+echo " \_____|_|  \___/ \_/ \___|_|   |_|  "
+echo " "
+
+# show some feedback on the console
+if [ -f $DEXTERSCRIPT/functions_library.sh ]; then
+  source $DEXTERSCRIPT/functions_library.sh
+  feedback "Welcome to GrovePi Installer."
 else
-	echo "Cloning"
-    cd $PIHOME/$DEXTER/
-    git clone https://github.com/DexterInd/GrovePi
-    cd GrovePi
+  echo "Welcome to GrovePi Installer."
+fi
+echo "Updating GrovePi for $selectedbranch branch with the following options:"
```

Comment:
these echo could totally be feedback instead

Reply:
See the other comment you left:
> This is silly, either functions_library.sh is used or it's not. Pull it in, curl it if necessary, and make use of it.